File: tests/mysql_test/compare.sql.php
<?
	
	require_once('db_mysql.class.php');
	
	set_time_limit(0);	

	$config = array(
					'sql_database' => 'test',
					'sql_user' => 'root',
					'sql_pass' => 'Dmitrik',
					'sql_tbl_prefix' => '',
					'debug' => 0,
					'hooks' => array()
					);

	$DB = new db_driver_mysql($config);
	$DB->connect();

	$arr = array(
					'date' => '2005-07-08',
					'operation' => 'mu'
				);

	$select = array(
					'date','operation'
					);
					
	for ($i=0;$i<100000;$i++)
	{
		$DB->select('log',$select,"`id`<20 or `operation`='mu'");
		$DB->queryCollect();
		//$DB->exec();	
		//$DB->getAssocArray();
		
		$DB->insert('log',$arr);
		$DB->queryCollect();
		//$DB->exec();
		
		$arr['operation'] = 'um';		
		$DB->update('log',$arr,'1');
		$arr['operation'] = 'mu';
		$DB->limit(50);
		$DB->queryCollect();
		//$DB->exec();	
	}
?>

File: tests/mysql_test/compare.sql.cc
#include <baseEx.h>
#include <dbMysql.h>

using namespace dodo;

using namespace std;

int main(int argc, char **argv)
{
	dbMysql pp;	
	try
	{
		pp.setDbInfo(string("test"),"",0,"","Dmitrik","/tmp/mysql.sock");
		
		pp.connect();		
		
		map<string,string> arr;
		arr["date"] = "2005-07-08";
		arr["operation"] = "mu";
		
		vector<string> select;
		select.push_back("date");
		select.push_back("operation");
		
		for (int i=0;i<100000;i++)
		{
			pp.select("log",select,"`id`<20 or `operation`='mu'");
			pp.queryCollect();
			//pp.exec();
			//pp.fetch();
			
			pp.insert("log",arr);
			pp.queryCollect();
			//pp.exec();
			
			arr["operation"] = "um";
			pp.update("log",arr);
			arr["operation"] = "mu";
			pp.limit(50);
			pp.queryCollect();
			//pp.exec();
		}
	}
	catch(baseEx ex)
	{
		printf("BASEexception was in: in %s at %ld\n",ex.file.c_str(),ex.line);
	}	
	return 0;
}
